### ISSUE TITLE
TextField multi-line hint text displays ellipsis by default

### ISSUE THREAD

**Issue by crasowas** [flutter/flutter#127477]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Use any TextField and set the `maxLines` to `null`.
2. Set a long enough hint text.

### Expected results

Multi-line hint text is displayed in full.

### Actual results

Multi-line hint text is truncated to display only one line.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Container(
            color: Colors.white,
            width: 300,
            height: 200,
            child: TextField(
              maxLines: null,
              decoration: InputDecoration(
                hintText:
                'Text that suggests what sort of input the field accepts.' * 5,
                // hintStyle: const TextStyle(overflow: TextOverflow.fade),
                border: InputBorder.none,
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

<img width="359" alt="Screenshot_1" src="https://github.com/flutter/flutter/assets/110958409/697de117-ba76-4bcc-b653-98ca5503e2c6">

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.1, on macOS 13.3.1 22E261 darwin-arm64, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.78.0)
[✓] Connected device (6 available)
[✓] Network resources
```

</details>

**Comment by huycozy** [comment]
Hi @crasowas 
Please correct me if I'm wrong. You can have the expected result if you set hintStyle's overflow as different than `ellipsis`, for eg: `hintStyle: const TextStyle(overflow: TextOverflow.visible)` 

I tried and see it can work.

<details>
<summary>Demo</summary>

![Screenshot 2023-05-24 at 19 18 24](https://github.com/flutter/flutter/assets/104349824/02a722ed-fd99-4e21-924f-a4e3c5686e90)
</details>

**Comment by crasowas** [comment]
@huycozy Thanks for your reply! The solution you provided is working.

This appears to be caused by `overflow` being set to `TextOverflow.ellipsis` by default.

[input_decorator.dart](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/input_decorator.dart)

```dart
final Widget? hint = hintText == null ? null : AnimatedOpacity(
  opacity: (isEmpty && !_hasInlineLabel) ? 1.0 : 0.0,
  duration: _kTransitionDuration,
  curve: _kTransitionCurve,
  child: Text(
    hintText,
    style: hintStyle,
    textDirection: decoration.hintTextDirection,
    overflow: hintStyle.overflow ?? TextOverflow.ellipsis,
    textAlign: textAlign,
    maxLines: decoration.hintMaxLines,
  ),
);
```

According to the documentation comments, the combination of `maxLines: null` and `TextOverflow.ellipsis` will result in only one line being displayed.

[text_painter.dart](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/text_painter.dart)

```dart
/// The string used to ellipsize overflowing text. Setting this to a non-empty
/// string will cause this string to be substituted for the remaining text
/// if the text can not fit within the specified maximum width.
///
/// Specifically, the ellipsis is applied to the last line before the line
/// truncated by [maxLines], if [maxLines] is non-null and that line overflows
/// the width constraint, or to the first line that is wider than the width
/// constraint, if [maxLines] is null. The width constraint is the `maxWidth`
/// passed to [layout].
///
/// After this is set, you must call [layout] before the next call to [paint].
///
/// The higher layers of the system, such as the [Text] widget, represent
/// overflow effects using the [TextOverflow] enum. The
/// [TextOverflow.ellipsis] value corresponds to setting this property to
/// U+2026 HORIZONTAL ELLIPSIS (…).
String? get ellipsis => _ellipsis;
```

Can we do better? For example, replace `overflow: hintStyle.overflow ?? TextOverflow.ellipsis,` with `overflow: hintStyle.overflow ?? (decoration.hintMaxLines == null ? null : TextOverflow.ellipsis)`, 
so that the multi-line hint text can be fully displayed by default.

**Comment by huycozy** [comment]
Thanks for your response, @crasowas 
What you mentioned above is similar to opinions at issue https://github.com/flutter/flutter/issues/104243. I would suggest you leave there your comment with above information will help to raise the issue.

I will close this issue in favor of #104243. Please follow up on it for further updates.

**Comment by crasowas** [comment]
@huycozy This issue [#104243](https://github.com/flutter/flutter/issues/104243) has been fixed in [#114378](https://github.com/flutter/flutter/pull/114378), and the current issue is a brand new one. 

I suggest closing issue [#104243](https://github.com/flutter/flutter/issues/104243) instead of the current issue.

**Comment by huycozy** [comment]
Hi @crasowas,
Thanks for your update. I closed https://github.com/flutter/flutter/issues/104243 as fixed by PR you mentioned.

I also re-open this issue for further discussion. 

>Can we do better? For example, replace overflow: hintStyle.overflow ?? TextOverflow.ellipsis, with overflow: hintStyle.overflow ?? (decoration.hintMaxLines == null ? null : TextOverflow.ellipsis),
so that the multi-line hint text can be fully displayed by default.

With regard to this, I think this would make the logic more complicated, and not sure what the affected scope of that would be. Instead, I see below code can work as expected:

<details>
<summary>Sample code (multi-line hint text is fully displayed)</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: Center(
          child: SizedBox(
            height: 200,
            child: TextField(
              maxLines: null,
              decoration: InputDecoration(
                hintText: 'Hello have a nice day. How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing?',
                hintMaxLines: null,
                hintStyle: TextStyle(overflow: TextOverflow.visible),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```
</details>


<details>
<summary>Demo</summary>

<img width="400" src="https://github.com/flutter/flutter/assets/104349824/bc23597f-4ba4-40b3-872e-29dd379cffaf">
</details>
